bitnami/jaeger/templates/_helpers.tpl

```diff
@@ -0,0 +1,121 @@
+{{/* vim: set filetype=mustache: */}}
+
+
+{{/*
+Return the proper jaeger image name
+*/}}
+{{- define "jaeger.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper cassandra external image name
+*/}}
+{{- define "jaeger.cassandra.cqlshImage" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.cassandra.cqlshImage "global" .Values.global) }}
+{{- end -}}
+
+
+{{/*
+Create the name of the query deployment
+*/}}
+{{- define "jaeger.query.fullname" -}}
+    {{ printf "%s-query" (include "common.names.fullname" .) }}
+{{- end -}}
+
+{{/*
+Create a container for checking cassandra availability
+*/}}
+{{- define "jaeger.waitForDBInitContainer" -}}
+- name: jaeger-cassandra-ready-check
+  image: {{ include "jaeger.cassandra.cqlshImage" . }}
+  command:
+    - /bin/bash
+  args:
+    - -ec
+    - |
+      #!/bin/bash
+
+      set -o errexit
+      set -o nounset
+      set -o pipefail
+
+      . /opt/bitnami/scripts/libos.sh
+
+      check_cassandra_keyspace_schema() {
+          echo "SELECT 1" | cqlsh -u $CASSANDRA_USERNAME -p $CASSANDRA_PASSWORD -e "SELECT COUNT(*) FROM ${CASSANDRA_KEYSPACE}.traces"
+      }
+
+      info "Connecting to the Cassandra instance $CQLSH_HOST:$CQLSH_PORT"
+      if ! retry_while "check_cassandra_keyspace_schema" 12 30; then
+        error "Could not connect to the database server"
+        exit 1
+      else
+        info "Connection check success"
+      fi
+  env:
+  - name: CQLSH
+    value: /opt/bitnami/cassandra/bin/cqlsh
+  - name: CQLSH_HOST
+    value: {{ printf "%s-cassandra" (include "common.names.fullname" .) }}
```

Comment:
What if we use an external Cassandra?

Reply:
The external cassandra image would have the same service name isn't it? Can't see the issue here

Reply:
Refactored [a6fa423](https://github.com/bitnami/charts/pull/13480/commits/a6fa42382e0213dcff79dc9fd247276a5ff21b08)

Reply:
No it wouldn't. You don't know the exact name of the external cassandra instance. Imagine it is a cassandra hosted in Azure, the hostname would be something like this

cassandra.database.windows.net

The same happens with the port